From 8323587f79179b0d4debd72320d71a78c7ab0a54 Mon Sep 17 00:00:00 2001
From: Bart Ribbers <bribbers@disroot.org>
Date: Wed, 25 Mar 2020 17:08:21 +0100
Subject: [PATCH] Remove libshadowutils-dev dependency

In response to https://git.sailfishos.org/mer-core/nemo-qml-plugin-systemsettings/merge_requests/3#note_42842
The USER environment variable is set after login, and since this package
is not used before login, it is safe to use. This way we don't have to
keep a downstream attempt to turn shadow in a shared library which no
one is going to upstream.
---
 rpm/nemo-qml-plugin-systemsettings.spec |  1 -
 src/developermodesettings.cpp           | 11 +----------
 src/src.pro                             |  2 +-
 3 files changed, 2 insertions(+), 12 deletions(-)

diff --git a/rpm/nemo-qml-plugin-systemsettings.spec b/rpm/nemo-qml-plugin-systemsettings.spec
index 15b0ecc..bb01ae9 100644
--- a/rpm/nemo-qml-plugin-systemsettings.spec
+++ b/rpm/nemo-qml-plugin-systemsettings.spec
@@ -22,7 +22,6 @@ BuildRequires:  pkgconfig(profile)
 BuildRequires:  pkgconfig(mce) >= 1.21.0
 BuildRequires:  pkgconfig(mlite5)
 BuildRequires:  pkgconfig(usb-moded-qt5)
-BuildRequires:  pkgconfig(libshadowutils)
 BuildRequires:  pkgconfig(blkid)
 BuildRequires:  pkgconfig(libcrypto)
 BuildRequires:  pkgconfig(nemodbus) >= 2.1.16
diff --git a/src/developermodesettings.cpp b/src/developermodesettings.cpp
index 4b8ce4a..010e191 100644
--- a/src/developermodesettings.cpp
+++ b/src/developermodesettings.cpp
@@ -41,9 +41,6 @@
 #include <QNetworkInterface>
 #include <transaction.h>
 
-#include <getdef.h>
-#include <pwd.h>
-
 /* Symbolic constants */
 #define PROGRESS_INDETERMINATE (-1)
 
@@ -113,13 +110,7 @@ DeveloperModeSettings::DeveloperModeSettings(QObject *parent)
     , m_localInstallFailed(false)
     , m_localDeveloperModePackagePath(get_cached_package(QStringLiteral("*")))  // Initialized to possibly incompatible package
 {
-    int uid = getdef_num("UID_MIN", -1);
-    struct passwd *pwd;
-    if ((pwd = getpwuid(uid)) != NULL) {
-        m_username = QString(pwd->pw_name);
-    } else {
-        qCWarning(lcDeveloperModeLog) << "Failed to return username using getpwuid()";
-    }
+    m_username = QString(qgetenv("USER"));
 
     // Resolve and update local package path
     if (!m_localDeveloperModePackagePath.isEmpty()) {


